external-crates/move/move-compiler/src/hlir/translate.rs

```diff
@@ -1957,3 +2008,40 @@ fn check_trailing_unit_statement(context: &mut Context, sp!(_, s_): &mut H::Stat
         S::Loop { block, .. } => check_trailing_unit(context, block),
     }
 }
+
+/// Generates warnings for unused struct fields.
+fn gen_unused_warnings(
+    context: &mut Context,
+    is_source_module: bool,
+    structs: &UniqueMap<StructName, H::StructDefinition>,
+) {
+    if !is_source_module {
+        // generate warnings only for modules compiled in this pass rather than for all modules
+        // including pre-compiled libraries for which we do not have source code available and
+        // cannot be analyzed in this pass
+        return;
+    }
+
+    for (_, sname, sdef) in structs {
+        context
+            .env
+            .add_warning_filter_scope(sdef.warning_filter.clone());
+
+        if let H::StructFields::Defined(fields) = &sdef.fields {
+            for (f, _) in fields {
+                if !context
+                    .used_fields
+                    .get(sname)
+                    .is_some_and(|names| names.contains(&f.value()))
+                {
+                    let msg = format!("The {} field of the {sname} type is unused", f.value());
```

Comment:
```suggestion
                    let msg = format!("The '{}' field of the '{sname}' type is unused", f.value());
```

Currently I try to wrap user specified code/input in single quotes `'`. I should really switch all of those to backticks `\`` but for now the single quote is consistent